### PR TITLE
Update some Rosetta error copy

### LINKED ISSUE #6932: Better error for missing data in the archive node during Rosetta query
Add more context to the block missing error here:

https://github.com/MinaProtocol/mina/pull/6926/files#diff-a522b82cff72b23f2e6d2a17999205389e7f2ff51ba3803d145a454b2d390c98R382

Specifically, return enough information for a user to realize they need to recover missing data in their archive node.

### PR DESCRIPTION
**Pull request by omerzach**
Thank you for contributing to Mina! Please see `CONTRIBUTING.md` if you haven't
yet. In that doc, there are more details around how to start our CI.

Just updating some Rosetta error strings. No special testing makes sense.

Checklist:

- [x] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [x] Tests were added for the new behavior
- [ ] All tests pass (CI will check this if you didn't)
- [x] Serialized types are in stable-versioned modules
- [x] Does this close issues? List them:

Closes #6932

### CODE REVIEW

File: src/lib/rosetta_lib/errors.ml [hunk 1]
```diff
@@ -121,21 +121,23 @@ end = struct
     | `Account_not_found addr ->
         Some
           (sprintf
-             !"You attempt to lookup %s but we couldn't find it in the ledger."
+             !"You attempted to lookup %s but we couldn't find it in the \
```
Comment (psteckler) [inline src/lib/rosetta_lib/errors.ml:124]: I would insert a comma before `but`

**Review by psteckler — APPROVED** [review]

File: src/lib/rosetta_lib/errors.ml (hunk 1 above)
Comment (omerzach) [reply src/lib/rosetta_lib/errors.ml:124]: Sure

Comment (omerzach) [reply src/lib/rosetta_lib/errors.ml]: Done, thanks

**Comment by bkase** [comment]
!ci-build-me

**Review by bkase — APPROVED** [review]

**Comment by bkase** [comment]
!ci-build-me

**Pull request merged**